plugins/reflection/src/main/java/org/qbicc/plugin/reflection/ReflectiveElementRegistry.java

```diff
@@ -127,6 +143,65 @@ public boolean isReflectiveMethod(String className, String methodName, MethodDes
         return false;
     }
 
+
+    /*
+     * Late phase mapping methods
+     */
+    public void registerReflectiveType(LoadedTypeDefinition ltd) {
+        reflectiveLoadedTypes.add(ltd);
+    }
+
+    public boolean isReflectiveType(LoadedTypeDefinition ltd) {
+        return reflectiveLoadedTypes.contains(ltd);
+    }
+
+    public boolean registerReflectiveMethod(MethodElement e) {
+        boolean added = reflectiveMethodElements.add(e);
+        if (added) {
+            ReachabilityRoots.get(ctxt).registerReflectiveEntrypoint(e);
+            ctxt.submitTask(e, methodElement -> {
+                Vm vm = ctxt.getVm();
+                VmThread thr = vm.newThread("genMethodAccessor", vm.getMainThreadGroup(), false,  Thread.currentThread().getPriority());
```

Comment:
I guess this is somehow happening during the earliest phases where we might not have a thread attached to the main thread?

Reply:
I'm taking a look. Might not actually be needed now.

I think it might actually have been the converse... some of this logic used to be run in an ADD postHook, so the ADD TaskWrapper didn't automatically attach the task to a thread.

Reply:
Yep...cargo cult programming.  Didn't need the attach here (or in several other places) got rid of them all in 5c3f3942